recipes/lora_finetune_single_device.py

```diff
@@ -219,6 +219,14 @@ def setup(self, cfg: DictConfig) -> None:
             ),
         )
 
+        self.gpu_monitor = None
+        if self._device.type == "cuda":
+            self.gpu_monitor = utils.memory.build_gpu_memory_monitor()
+        self.tps = 0
+        self.mem_pct = 0
+        self.mem_cache_retries = 0
```

Comment:
I am not sure if we need these variables to have "self.". Maybe we can just initialized them in train() instead? They are not used anywhere else, as far as I can tell